DOL.WHD.Section14c.PdfAPi/PdfHelper/PdfHelper.cs

```diff
@@ -0,0 +1,274 @@
+               
+                foreach (HtmlNode node in tdNodes)
+                {
+                    node.Remove();
+                }
+
+                tempString = doc.DocumentNode.InnerHtml;
+            }
+            return tempString;
+        }
+
+        /// <summary>
+        /// Concatenate PDF Document By Array
+        /// </summary>
+        /// <param name="documentContentByteArray"></param>
+        /// <returns></returns>
+        public static PdfDocument ConcatenatePDFDocumenByArray(PdfDocument outputDocument, byte[] buffer)
+        {
+            if (buffer == null)
+                throw new ArgumentException("No resource");
+
+            using (var stream = new MemoryStream(buffer))
+            {     
+                // Open the document to import pages from it.
+                var inputDocument = PdfReader.Open(stream, PdfDocumentOpenMode.Import);
+
+                // Iterate pages
+                int count = inputDocument.PageCount;
+                for (int idx = 0; idx < count; idx++)
+                {
+                    // Get the page from the external document...
+                    PdfPage page = inputDocument.Pages[idx];
+                   
+                    // ...and add them twice to the output document.
+                    outputDocument.AddPage(page);
+                }
+            }
+            return outputDocument;
+        }
+
+        /// <summary>
+        /// Create PDF from Image
+        /// </summary>
+        /// <param name="outputDocument"></param>
+        /// <param name="buffer"></param>
+        /// <returns></returns>
+        public static PdfDocument ConcatenatePDFWithImage(PdfDocument outputDocument, byte[] buffer)
+        {
+            if (buffer == null)
+                throw new ArgumentException("No resource");
+
+            using (var stream = new MemoryStream(buffer))
+            {
+                // Create an empty page
+                // Get an XGraphics object for drawing
+                XGraphics gfx = XGraphics.FromPdfPage(outputDocument.AddPage());
+
+                Image image = Image.FromStream(stream);
+
+                XImage img = XImage.FromGdiPlusImage(image);
+                gfx.DrawImage(img, 0, 0);
+            }
+            return outputDocument;
+        }        
+
+        /// <summary>
+        /// Concatenate PDF Document By File Path
+        /// </summary>
+        /// <param name="filePaths"></param>
+        /// <returns></returns>
+        public static PdfDocument ConcatenatePDFDocumentByPath(PdfDocument outputDocument, List<string> filePaths)
+        {
+            // Iterate files
+            foreach (var file in filePaths)
+            {
+                var extension = Path.GetExtension(file);
+
+                var regex = new Regex(Constants.PdfConcatenateSupportedFileTypes);
```

Comment:
The solution is currently only support images, and pdf. I did not want to make it configurable, just in case someone want to add unsupported types. for example doc or docx.

Reply:
Oooh, that makes total sense. Thanks for explaining!